File: src/main/java/io/github/takenoha/towerdefense/paper/VoucherEntityPolicy.java
package io.github.takenoha.towerdefense.paper;

/** Pure entity-boundary rules for protecting voucher items. */
public final class VoucherEntityPolicy {
    private VoucherEntityPolicy() {
    }

    public static boolean blocksInteraction(boolean handIsVoucher, boolean entityContainsVoucher) {
        return handIsVoucher || entityContainsVoucher;
    }

    public static boolean blocksHangingBreak(boolean entityContainsVoucher) {
        return entityContainsVoucher;
    }
}

File: src/main/kotlin/io/github/takenoha/towerdefense/paper/VoucherEntityPolicy.kt
package io.github.takenoha.towerdefense.paper

/** Pure entity-boundary rules for protecting voucher items. */
class VoucherEntityPolicy private constructor() {
    companion object {
        @JvmStatic
        fun blocksInteraction(handIsVoucher: Boolean, entityContainsVoucher: Boolean): Boolean =
            handIsVoucher || entityContainsVoucher

        @JvmStatic
        fun blocksHangingBreak(entityContainsVoucher: Boolean): Boolean = entityContainsVoucher
    }
}
